background/rust.tex

```diff
@@ -11,18 +11,33 @@ \section{The Rust Programming Language} % (fold)
 
-\subsubsection{The Core Library}
+\subsubsection{The Rust Standard Library}
+\label{sec:rsl}
 
-\rust's {\core} library, defines the platform independent part of the language.
-This library is not directly accessible to the programmer, instead its stable parts are reexported through {\rust}'s standard library called {\std}.
-In this way, {\core} is not interfaced with directly in normal {\rust} programs, so it can be modified while {\std}'s interface stays the same.
-The standard library also reexports a handful of other libraries and does the task of implementing platform-dependent features for all of its target \glspl{os}.
+The \gls{rsl} is included into all {\rust} programs and strives to be an abstraction layer to write portable applications.
+The library is implemented as a facade, meaning that the crate {\std} provides a thin wrapper which reexports a stable interface for crates like \lib{alloc} (for memory allocation), \lib{collections} (general purpose data structures) and {\core} (described in the next section).
+In addition to providing the facade, {\std} crate defines platform abstractions like files, network protocols and threading, all of which depends on an OS.
+This basic fact renders the {\std} crate unusable in an embedded system as these dependencies will not be met.
+Luckly it is possible to opt-out of using the library.
 
-\autoref{tab:rust:datatypes} lists the primitive datatypes available in {\rust}, the {\core} library defines many functions and abstractions that are used to manage and manipulate these datatypes.
-Most of the datatypes are self explanatory when it comes to representation and functionality, the ones that are noticeably different from {\C} are discussed throughout the following sections.
+It is important to notice the distinction between the \gls{rsl} which includes the functionality provided by the underlying Collection, Allocation and Core libraries, and the {\std} crate which defines the facade and the mentioned additional abstractions.
+
+\subsubsection{The Rust Core Library}
+\label{sec:rcl}
+
+\gls{rcl}, defines the platform independent part of the language.
+This library is not intended to be directly interfaced with by the application programmer, instead a stable interface is reexported through \gls{rsl}.
+In this way, \gls{rcl} is not interfaced with directly in normal {\rust} programs.
+This gives the developer the ability to modify the library while the \gls{rsl} interface stays the same.
+%The standard library also reexports a handful of other libraries and does the task of implementing platform-dependent features for all of its target \glspl{os}.
+
+The distinction between \gls{rcl} and the {\core} crate is not that clear as the {\std} and \gls{rsl} as \gls{rcl} does not export any functionallity which is not defined in {\core}.
```

Comment:
Denne setningen er kanskje litt rar, kan kanske endres til dette: `the distinction between RCL and the core crate is not as clear as the distinction between std and RSL, as RCL does not export any functionality which is not defined in core.`

Også, når det nevnes at "this distinction is not as clear" så burde det kanskje nevnes _hva_ forskjellen er?

Tror du den er nødvendig i det hele tatt?